### PR TITLE
[NP-3341] Use ProxySlot on WizardSection isAvailable

### PR DESCRIPTION
**Pull request by KernelDeimos**

### CODE REVIEW

File: src/foam/u2/wizard/WizardletSection.js [hunk 1]
```diff
@@ -79,6 +79,17 @@ foam.CLASS({
   ],
 
   methods: [
+    function activateProxySlot() {
+      var proxy$ = foam.core.ProxySlot.create();
+      this.isAvailable$ = proxy$;
+      var f = () => {
+        var isAvailable$ = this.section.createIsAvailableFor(this.data$);
```
Comment (kgrgreer) [inline src/foam/u2/wizard/WizardletSection.js:86]: Could these two lines be combined?

File: src/foam/u2/wizard/WizardletSection.js [hunk 2]
```diff
@@ -79,6 +79,17 @@ foam.CLASS({
...
+        proxy$.delegate = isAvailable$;
+      };
+      this.data$.sub(f);
```
Comment (kgrgreer) [inline src/foam/u2/wizard/WizardletSection.js]: Do you need to wrap this in an this.onDetach()?

File: src/foam/u2/wizard/WizardletSection.js (hunk 1 above)
Comment (kgrgreer) [reply src/foam/u2/wizard/WizardletSection.js:86]: Shouldn't createIsAvailableFor(slot) be listening for slot changes? Why do we need to use the ProxySlot?

Comment (KernelDeimos) [reply src/foam/u2/wizard/WizardletSection.js:86]: I'll be honest, after several hours of debugging this was the best solution I could come up with. I have no idea why what you described doesn't work - I tried that first.

File: src/foam/u2/wizard/WizardletSection.js (hunk 2 above)
Comment (KernelDeimos) [reply src/foam/u2/wizard/WizardletSection.js]: yes

File: src/foam/u2/wizard/WizardletSection.js (hunk 1 above)
Comment (KernelDeimos) [reply src/foam/u2/wizard/WizardletSection.js:86]: It's possible that the slot gets broken, but I tried adding `} catch (e) { console.error(e);` to the try blocks in Slot.js and I didn't see any exceptions get thrown.

**Comment by kgrgreer** [comment]
I'll merge, but I think there must be an underlying bug that is preventing a simpler solution from working. Please schedule some time with me to debug.

**Pull request merged**